test/inductor/test_aot_inductor.py

```diff
@@ -187,6 +187,23 @@ def forward(self, x, y):
         )
         self.assertTrue(actual_path == expected_path)
 
+    def test_constant_folding(self):
+        class Model(torch.nn.Module):
+            def __init__(self, device):
+                super().__init__()
+                self.w_pre = torch.randn(4, 4, device=device)
+                self.b = torch.randn(4, device=device)
+
+            def forward(self, x):
+                w_transpose = torch.transpose(self.w_pre, 0, 1)
+                w_relu = torch.nn.functional.relu(w_transpose)
+                w = w_relu + self.b
+                return torch.matmul(x, w)
+
+        example_inputs = (torch.randn(4, 4, device=self.device),)
+        with config.patch({"aot_inductor.use_runtime_constant_folding": True}):
```

Comment:
I guess we would need to add the pybind, but I would say we should keep that away too since it's not directly related to this PR. (And as you mentioned it's a big PR already...)